inc/saimymac.h

```diff
@@ -0,0 +1,319 @@
+/**
+ * Copyright (c) 2014 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saimymac.h
+ *
+ * @brief   This module defines SAI My MAC
+ */
+
+#if !defined (__SAIMYMAC_H_)
+#define __SAIMYMAC_H_
+
+#include <saitypes.h>
+
+/**
+ * @brief My MAC entry attribute IDs
+ */
+typedef enum _sai_my_mac_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_MY_MAC_ATTR_START,
+
+    /**
+     * @brief Priority
+     *
+     * Value must be in the range defined in
+     * \[#SAI_MY_MAC_TABLE_ATTR_MINIMUM_PRIORITY,
+     * #SAI_MY_MAC_TABLE_ATTR_MAXIMUM_PRIORITY\]
+     * (default = #SAI_MY_MAC_TABLE_ATTR_MINIMUM_PRIORITY)
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_MY_MAC_ATTR_PRIORITY = SAI_MY_MAC_ATTR_START,
+
+    /**
+     * @brief Object id of the My MAC table
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_MY_MAC_TABLE
+     */
+    SAI_MY_MAC_ATTR_TABLE_ID,
+
+    /**
+     * @brief Associated Port, LAG object id,
+     * if not specified any port will match
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_PORT, SAI_OBJECT_TYPE_LAG
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_MY_MAC_ATTR_PORT_ID,
+
+    /**
+     * @brief Associated Vlan Id,
+     * if not specified any vlan id will match
+     *
+     * @type sai_uint16_t
+     * @flags CREATE_ONLY
+     * @isvlan true
+     * @default 0
+     */
+    SAI_MY_MAC_ATTR_VLAN_ID,
+
+    /**
+     * @brief MAC Address
+     *
+     * @type sai_mac_t
+     * @flags CREATE_ONLY
+     * @default vendor
+     */
+    SAI_MY_MAC_ATTR_MAC_ADDRESS,
+
+    /**
+     * @brief MAC Address Mask
+     *
+     * @type sai_mac_t
+     * @flags CREATE_ONLY
+     * @default vendor
+     */
+    SAI_MY_MAC_ATTR_MAC_ADDRESS_MASK,
+
+    /**
+     * @brief Enable IPv4 processing
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default true
+     */
+    SAI_MY_MAC_ATTR_V4_ENABLE,
```

Comment:
Since V4 and V6 enable flags are in the RIF as well, please clarify what would be the expected behavior in case of a conflict between my_mac and rif flags.

Reply:
My MAC entry determines if packet gets L3 forwarded or L2 forwarded.
Once the packet is in the L3 forwarding pipeline, RIF L3 admin state functionality
will take effect as it does currently.